### PR TITLE
Jenkins

### PR DESCRIPTION
**Pull request by rowleya**
Fixes for integration tests.  Fixes the spike handling in the presence of direct rows, and allows things to work with pre-seeded RNGs.

### CODE REVIEW

File: neural_modelling/src/neuron/spike_processing.c [hunk 1]
```diff
@@ -63,69 +63,72 @@ static inline void _do_direct_row(address_t row_address) {
...
+    // Is there another address in the population table?
+    // Note, this is fairly quick to check, so leave interrupts disabled
+    } else if (population_table_get_next_address(
+            row_address, n_bytes_to_transfer)) {
+        something_to_do = true;
+    } else {
+
+        // Are there any more spikes to process?
+        while (!something_to_do && in_spikes_get_next_spike(&spike)) {
+
```
[the remark this answers is not in the document]
Comment (rowleya) [reply neural_modelling/src/neuron/spike_processing.c]: Except that something_to_do is also used to get you out of the loop before there are no spikes i.e. if there is a transfer to be done.

File: spynnaker/pyNN/models/neural_projections/connectors/fixed_number_post_connector.py [hunk 2]
```diff
@@ -156,7 +156,7 @@ def get_n_connections_to_post_vertex_maximum(self):
         n_connections = utility_calls.get_probable_maximum_selected(
             self._n_post_neurons * self._n_pre_neurons,
             self._n_post * self._n_pre_neurons, selection_prob,
-            chance=1.0/10000.0)
+            chance=1.0/100000.0)
         return int(math.ceil(n_connections))
```
[the remark this answers is not in the document]
Comment (rowleya) [reply spynnaker/pyNN/models/neural_projections/connectors/fixed_number_post_connector.py]: I'm not sure why this was needed, or why the chance was suddenly needed in that location (the default is 1/100 I think).  We could make them match for consistency.

File: neural_modelling/src/neuron/spike_processing.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (andrewgait) [reply neural_modelling/src/neuron/spike_processing.c]: Oops, must have missed that - guess I missed it in the formatting somewhere.  In that case it's fine.

File: spynnaker/pyNN/models/neural_projections/connectors/fixed_number_post_connector.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (andrewgait) [reply spynnaker/pyNN/models/neural_projections/connectors/fixed_number_post_connector.py]: Sounds good to me.

**Pull request merged**